dags/process_elife_zip_dag.py

```diff
@@ -156,17 +160,38 @@ def update_tiff_references_to_jpeg_in_article(**context):
             Key=s3_key,
             Body=etree.tostring(article_xml, xml_declaration=True, encoding='UTF-8')
         )
-        logger.info('%s uploaded to %s/%s', s3_key, DESTINATION_BUCKET, folder_name)
+        logger.info('%s uploaded to %s', s3_key, DESTINATION_BUCKET)
+
+    return s3_key
+
+
+def strip_related_article_tags_from_article_xml(**context) -> str:
+    s3_key = get_article_s3_key_from_previous_task(context)
+
+    s3 = get_aws_connection('s3')
+    response = s3.get_object(Bucket=DESTINATION_BUCKET, Key=s3_key)
+    article_xml = etree.parse(BytesIO(response['Body'].read()))
+
+    matches = article_xml.xpath('//related-article')
+    if matches:
+        for element in matches:
+            element.getparent().remove(element)
+
+        # upload modified document
+        s3_key = s3_key.replace('.xml', '-remove_related_article.xml')
```

Comment:
this is getting unwieldy, so we need to think a standard solution for names, what about an incremental number like `-edit_1`, `edit_2` so that it's clear in what order edits were performed?

Reply:
I like the idea of a better naming convention but I don't think `edit_1`, `edit_2` is very descriptive.

Another solution is, instead of downloading and uploading edits to s3, the article bytes could be passed from one task to the next via the return statement? This would result in only the original article being in s3 and the final xml being sent to the content-store.

Reply:
I think that solution would work well even with multiple workers (it's Airflow that passes things around for us) so interesting unless it creates tons of database rows as a result. We would reserve external storage like S3 for binary files whose size is on the order of MBs like images I guess.